lib/ecto/adapters/sql.ex

```diff
@@ -922,7 +926,7 @@ defmodule Ecto.Adapters.SQL do
 
     all_params = placeholders ++ Enum.reverse(params, conflict_params)
 
-    %{num_rows: num, rows: rows} = query!(adapter_meta, sql, all_params, opts)
+    %{num_rows: num, rows: rows} = query!(adapter_meta, sql, all_params, put_source(opts, schema_meta))
```

Comment:
I think since it's already being pattern matched out above we could do this?

```elixir
put_source(opts, source)

defp put_source(opts, source) when is_binary(source) do
  ...
end
```

Reply:
Sounds good to me.